### ISSUE TITLE
Feature_request: configure the number of packets sent

### ISSUE THREAD

**Issue by MiniPierre** [prometheus/blackbox_exporter#629]
Hello,

I am currently migrating from Smokeping to Blackbox-exporter, but I regret the fact that it is not possible to configure the number of packets sent each time, which was possible with Smokeping in order to perform quick successive queries. Is it possible to implement such feature ?

**Comment by SuperQ** [comment]
The blackbox_exporter is not a good replacement for smokeping.

You might be interested in the [smokeping_prober](https://github.com/SuperQ/smokeping_prober). :smile:

**Comment by SuperQ** [comment]
The blackbox_exporter, when doing ICMP, only ever sends one packet. So the equivalent for "number of packets" is based on the `probe_success` metric.

* The number of packets: `count_over_time(probe_success[5m])`
* The number of returned packets: `sum_over_time(probe_success[5m])`
* The availability: `avg_over_time(probe_success[5m])`

**Comment by MiniPierre** [comment]
Pushing your own project, nice idea ! :laughing: 

Actually, blackbox_exporter perfectly suits me need, except for this feature (which is not an absolute precondition), but I'll have a look at smokeping_prober :+1:

**Comment by SuperQ** [comment]
I work on both projects. One just happens to be "official".

I've used the blackbox_exporter exactly as you're suggesting, as a smokeping replacement. The problems with using the blackbox_exporter this way is what inspired the other project. Right tool for the right job.

The blackbox_exporter intentionally does things very simply, adding multi-ICMP packet would complicate it more than we want here.

This was previously discussed in https://github.com/prometheus/blackbox_exporter/issues/115. Closing this as duplicate.
